### PR TITLE
PLT-665:Added support for Kubernetes distro for cluster group

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: spectrocloud/resource_cluster_group.go [hunk 1]
```diff
@@ -88,6 +88,14 @@ func resourceClusterGroup() *schema.Resource {
 							Optional: true,
 							Default:  "",
 						},
+						"kubernetes_distro": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "k3s",
+							ForceNew:     true,
+							ValidateFunc: validation.StringInSlice([]string{"k3s", "cncf_k8s"}, false),
+							Description:  "The Kubernetes distro, allowed distribution are `k3s` and `cncf_k8s`.",
```
Comment (nikchern) [inline spectrocloud/resource_cluster_group.go:97]:
```suggestion
							Description:  "The Kubernetes distribution, allowed values are `k3s` and `cncf_k8s`.",
```

File: spectrocloud/resource_cluster_group.go [hunk 2]
```diff
@@ -88,6 +88,14 @@ func resourceClusterGroup() *schema.Resource {
 							Optional: true,
 							Default:  "",
 						},
+						"kubernetes_distro": {
```
Comment (nikchern) [inline spectrocloud/resource_cluster_group.go:91]: May be rename to "k8s_distribution" ? @SivaanandM

File: spectrocloud/resource_cluster_group.go [hunk 3]
```diff
@@ -184,6 +192,7 @@ func flattenClusterGroup(clusterGroup *models.V1ClusterGroup, d *schema.Resource
 						"memory_in_mb":             limitConfig.MemoryMiB,
 						"storage_in_gb":            limitConfig.StorageGiB,
 						"oversubscription_percent": limitConfig.OverSubscription,
+						"kubernetes_distro":        clusterConfig.KubernetesDistroType,
```
Comment (nikchern) [inline spectrocloud/resource_cluster_group.go:195]: see comment above.

File: spectrocloud/resource_cluster_group.go [hunk 4]
```diff
@@ -282,6 +291,7 @@ func toClusterGroup(c *client.V1Client, d *schema.ResourceData) *models.V1Cluste
 	var values string
 	resourcesObj, ok := d.GetOk("config")
 	endpointType := "Ingress" // default endpoint type is ingress
+	k8Distro := "k3s"
```
Comment (nikchern) [inline spectrocloud/resource_cluster_group.go:294]: default is already in the schema. Why it's needed here?

**Review by nikchern — APPROVED** [review]

File: spectrocloud/resource_cluster_group.go (hunk 2 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_group.go]: done

File: spectrocloud/resource_cluster_group.go (hunk 3 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_group.go]: addressed

File: spectrocloud/resource_cluster_group.go (hunk 4 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_group.go]: changed it

**Pull request merged**